lib/make-spawn-args.js

```diff
@@ -14,8 +13,20 @@ const makeSpawnArgs = options => {
     cmd,
     args,
     stdioString,
+    nodeGyp,
   } = options
 
+  if (nodeGyp) {
+    // npm already pulled this from env and passes it in to options
+    npm_config_node_gyp = nodeGyp
+  } else if (env.npm_config_node_gyp) {
+    // legacy mode for standalone user
+    npm_config_node_gyp = env.npm_config_node_gyp
+  } else {
+    // default
+    npm_config_node_gyp = require.resolve('node-gyp/bin/node-gyp.js')
```

Comment:
Even if we built a safeguard in here in case `require.resolve` wasn't available we'd still need this to throw if this default was needed and failed, so it's effectively a NOP.  If you need this code to work in an environment that doesn't have `require.resolve` you can pass in the option or set the environment variable.